go/ql/lib/semmle/go/security/UnicodeBypassValidationQuery.qll

```diff
@@ -0,0 +1,149 @@
+/**
+ * Provides a taint-tracking configuration for detecting "Unicode transformation mishandling" vulnerabilities.
+ */
+
+private import go
+import semmle.go.Concepts
+import semmle.go.dataflow.DataFlow
+import semmle.go.dataflow.TaintTracking
+import UnicodeBypassValidationCustomizations::UnicodeBypassValidation
+
+/** A state signifying that a logical validation has not been performed. */
+class PreValidation extends DataFlow::FlowState {
+  PreValidation() { this = "PreValidation" }
+}
+
+/** A state signifying that a logical validation has been performed. */
+class PostValidation extends DataFlow::FlowState {
+  PostValidation() { this = "PostValidation" }
+}
+
+private predicate intCheck(DataFlow::Node g, Expr e, boolean outcome) {
+  exists(DataFlow::CallNode cn, DataFlow::EqualityTestNode etn |
+    g = etn and
+    DataFlow::localFlow(cn.getResult(), etn.getAnOperand()) and
+    cn.getTarget()
+        .hasQualifiedName("strings",
+          [
+            "Index", "IndexAny", "IndexByte", "IndexFunc", "IndexRune", "LastIndex", "LastIndexAny",
+            "LastIndexByte", "LastIndexFunc", "Count",
+          ]) and
+    cn.getArgument(0).asExpr() = e and
+    etn.getAnOperand().getIntValue() = -1 and
```

Comment:
Comparing `Count` with `-1` doesn't sound likely!

Reply:
I would believe against `0` is ok, commit  https://github.com/github/codeql/pull/12994/commits/e0761a75fa209a56eda13bf394831c177d2eb930.